pyquil/api/_quantum_computer.py

```diff
@@ -154,40 +131,87 @@ def run(self, executable: Executable,
+
+        In the case of a single qubit i,j \in [0,1] then:
+        there is no readout error if p_{0|0} = p_{1|1} = 1.
+        the readout error is symmetric if p_{0|0} = p_{1|1} = 1 - epsilon.
+        the readout error is asymmetric if p_{0|0} != p_{1|1}.
+
+        If your quantum computer has this kind of asymmetric readout error then
+        `qc.run_symmetrized_readout` will symmetrize the readout error.
+
+        The readout error above is only asymmetric on a single bit. In practice the confusion
+        matrix on n bits need not be symmetric, e.g. for two qubits p_{ij|ij} != 1 - epsilon for
+        all i,j. In these situations a more sophisticated means of symmetrization is needed; and
+        we use orthogonal arrays (OA) built from Hadamard matrices.
+
+        The symmetrization types are specified by an int; the types available are:
+        -1 -- exhaustive symmetrization uses every possible combination of flips
+         0 -- trivial that is no symmetrization
+         1 -- symmetrization using an OA with strength 1
+         2 -- symmetrization using an OA with strength 2
+         3 -- symmetrization using an OA with strength 3
+        In the context of readout symmetrization the strength of the orthogonal array enforces
+        the symmetry of the marginal confusion matrices.
+
+        By default a strength 3 OA is used; this ensures expectations of the form <b_k * b_j * b_i>
+        for bits any bits i,j,k will have symmetric readout errors. Here expectation of a random
+        variable x as is denote <x> = sum_i Pr(i) x_i. It turns out that a strength 3 OA is also a
+        strength 2 and strength 1 OA it also ensures <b_j * b_i> and <b_i> have symmetric readout
+        errors for any bits b_j and b_i.
+
+        :param program: The program to run symmetrized readout on.
+        :param trials: The minimum number of times to run the program; it is recommend that this
+            number should be in the hundreds or thousands. This parameter will be mutated if
+            necessary.
+        :param symm_type: the type of symmetrization
+        :param meas_qubits: An advanced feature. The groups of measurement qubits. Only these
+            qubits will be symmetrized over, even if the program acts on other qubits.
+        :return: A numpy array of shape (trials, len(ro-register)) that contains 0s and 1s.
         """
-        Run a quil program in such a way that the readout error is made collectively symmetric
+        if type(symm_type) is not int:
+            raise ValueError("Symmetrization options are indicated by an int. See "
+                             "the docstrings for more information.")
 
-        This means the probability of a bitstring ``b`` being mistaken for a bitstring ``c`` is
-        the same as the probability of ``not(b)`` being mistaken for ``not(c)``
+        if meas_qubits is None:
+            meas_qubits = list(program.get_qubits())
```

Comment:
just double checking that this makes sense for the `None` case. if we pass a program with no MEASUREs and use symm_type = 0, then `run_symmetrized_readout` will append a DECLARE statement and MEASUREs, necessarily making it different from `run`

Reply:
Yes. It should be like `run_and_measure`.